linera-core/src/unit_tests/test_helpers.rs

```diff
@@ -1,57 +1,90 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-#[cfg(test)]
-use assert_matches::assert_matches;
 use linera_chain::{ChainError, ChainExecutionContext};
 use linera_execution::{ExecutionError, SystemExecutionError};
 
 use crate::{client::ChainClientError, local_node::LocalNodeError, worker::WorkerError};
 
-#[cfg(test)]
+/// Asserts that an error is due to insufficient funding during an operation.
 pub fn assert_insufficient_funding_during_operation<T>(
     obtained_error: &Result<T, ChainClientError>,
     operation_index: u32,
 ) {
-    assert_matches!(
-        obtained_error.as_ref().err().unwrap(),
+    let error = match obtained_error.as_ref().err().unwrap() {
         ChainClientError::LocalNodeError(LocalNodeError::WorkerError(
-            WorkerError::ChainError(error)
-        )) if matches!(&**error, ChainError::ExecutionError(
-            execution_error, ChainExecutionContext::Operation(index)
-        ) if index == &operation_index && matches!(**execution_error, ExecutionError::SystemError(
-            SystemExecutionError::InsufficientFunding { .. }
-        )))
-    );
+            WorkerError::ChainError(error),
+        )) => error,
+        _ => panic!("Expected a ChainClientError::LocalNodeError with a WorkerError::ChainError"),
+    };
+
+    let (execution_error, context) = match &**error {
+        ChainError::ExecutionError(execution_error, context) => (execution_error, context),
+        _ => panic!("Expected a ChainError::ExecutionError"),
+    };
```

Comment:
Idea: these `match` statements that either extract contents or `panic` could be replaced with `let ... = error else { panic!(); };`

```suggestion
    let ChainError::ExecutionError(execution_error, context) = error {
        panic!("Expected a `ChainError::ExecutionError`, found: {error:#?}");
    };
```